java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -232,44 +264,27 @@ protected abstract T deserializeField(
         return Something.wrap(this.deserializeField(context, field));
     }
 
-    @Override
-    public final int hashCode() {
-        return this.precomputedHashCode;
-    }
-
-    @Override
-    public final boolean equals(Object obj) {
-        if (this == obj) {
-            return true;
-        } else if (obj == null) {
-            return false;
-        } else if (this.getClass() != obj.getClass()) {
-            return false;
-        } else {
-            BondType<?> that = (BondType<?>) obj;
-            if (this.precomputedHashCode != that.precomputedHashCode) {
-                return false;
-            } else {
-                return this.equalsInternal(that);
-            }
-        }
-    }
-
-    // helper for subclasses to compute hash code
-    static int multiplyAndShift(int value, int n) {
-        return (value * n) ^ (value >>> (32 - n));
+    /**
+     * Helper for container subclasses to compute hash code as a value multipled by
+     * a number and then having its bits shifted to the right by a certain number.
+     * @param value a value
+     * @param multiplier multiplier
+     * @param bitsShifted number of bits shifted
+     * @return result
+     */
+    static int multiplyAndShiftForHashCodeComputation(int value, int multiplier, int bitsShifted) {
```

Comment:
It's a function to help with hashcode computation. The caller may want to specify those for different hash behavior.

---
In reply to: [124084837](https://github.com/Microsoft/bond/pull/502#discussion_r124084837) [](ancestors = 124084837)